packages/components/src/unit-control/index.tsx

```diff
@@ -154,87 +155,22 @@ function UnforwardedUnitControl(
 		setUnit( nextUnitValue );
 	};
 
-	const mayUpdateUnit = ( event: SyntheticEvent< HTMLInputElement > ) => {
-		if ( ! isNaN( Number( event.currentTarget.value ) ) ) {
-			refParsedQuantity.current = undefined;
-			return;
-		}
-		const [
-			validParsedQuantity,
-			validParsedUnit,
-		] = getValidParsedQuantityAndUnit(
-			event.currentTarget.value,
-			units,
-			parsedQuantity,
-			unit
-		);
-
-		refParsedQuantity.current = validParsedQuantity;
-
-		if ( isPressEnterToChange && validParsedUnit !== unit ) {
-			const data = Array.isArray( units )
-				? units.find( ( option ) => option.value === validParsedUnit )
-				: undefined;
-			const changeProps = { event, data };
-
-			onChangeProp?.(
-				`${ validParsedQuantity ?? '' }${ validParsedUnit }`,
-				changeProps
-			);
-			onUnitChange?.( validParsedUnit, changeProps );
-
-			setUnit( validParsedUnit );
-		}
-	};
-
-	const handleOnBlur: FocusEventHandler< HTMLInputElement > = mayUpdateUnit;
-
-	const handleOnKeyDown = ( event: KeyboardEvent< HTMLInputElement > ) => {
-		const { key } = event;
-		if ( key === 'Enter' ) {
-			mayUpdateUnit( event );
-		}
-	};
-
-	/**
-	 * "Middleware" function that intercepts updates from InputControl.
-	 * This allows us to tap into actions to transform the (next) state for
-	 * InputControl.
-	 *
-	 * @param  state  State from InputControl
-	 * @param  action Action triggering state change
-	 * @return The updated state to apply to InputControl
-	 */
-	const unitControlStateReducer: StateReducer = ( state, action ) => {
-		const nextState = { ...state };
-
-		/*
-		 * On commits (when pressing ENTER and on blur if
-		 * isPressEnterToChange is true), if a parse has been performed
-		 * then use that result to update the state.
-		 */
-		if ( action.type === inputControlActionTypes.COMMIT ) {
-			if ( refParsedQuantity.current !== undefined ) {
-				nextState.value = (
-					refParsedQuantity.current ?? ''
-				).toString();
-				refParsedQuantity.current = undefined;
+	let handleOnKeyDown;
+	if ( ! disableUnits && isUnitSelectTabbable && units.length ) {
+		handleOnKeyDown = ( event: KeyboardEvent< HTMLInputElement > ) => {
+			props.onKeyDown?.( event );
+			// Does the key match the first character of any units?
+			if ( reFirstCharacterOfUnits.test( event.key ) ) {
+				// Moves focus to the UnitSelectControl.
+				refInputSuffix.current?.focus();
```

Comment:
We will also need to assess if moving focus this way works well for folks relying on assistive technology

Reply:
Wow, I played around with it and I'm tempted to say it's acceptable, despite the focus moving 🤔 I think a big reason for that is because it's only moving to the next tabbable, and also the "type in the unit in the number field" is a hidden feature that only advanced users know in the first place. (Do we know how often people use this feature? I don't think I would ever discover it.)

But ideally, if we could accomplish this without a focus jump that might be even better. I guess we can't do something like remember the successive keydowns and wait for a complete match, because there could be units like `m` and `mm` ☹️

Reply:
I also found this to be quite an intuitive experience despite the focus moving. I'm happy though to defer to others who have greater experience with these components and accessibility requirements.

The point regarding this hidden feature not being that well known is a good one.

Reply:
> I'm not sure how feasible this is without introducing a lot of complexity to a component that is quite complex already

Agreed. Given that everyone here thinks the behavior is intuitive, let's move forward with the current implementation 👍

Reply:
>But ideally, if we could accomplish this without a focus jump that might be even better.

For posterity, I can verify that it can work but does add some complexity. It was how I first approached this and I've since revisited it because I realized that it already enables pasting so it might have a comparable level of complexity to this approach with `onPaste` handling added. Even with some success in simplifying it in my latest attempt it comes out looking complex compared to this. I have an inkling that it could be simplified further but would first require some overhaul to `InputControl` to allow for more flexible extensibility.